javax-money/src/main/java/com/fasterxml/jackson/datatype/money/CurrencyUnitDeserializer.java

```diff
@@ -0,0 +1,32 @@
+package com.fasterxml.jackson.datatype.money;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.jsontype.TypeDeserializer;
+import org.apiguardian.api.API;
+
+import javax.money.CurrencyUnit;
+import javax.money.Monetary;
+import java.io.IOException;
+
+import static org.apiguardian.api.API.Status.MAINTAINED;
+
+@API(status = MAINTAINED)
+public final class CurrencyUnitDeserializer extends JsonDeserializer<CurrencyUnit> {
```

Comment:
Usually should extend either `StdDeserializer` or (if applicable) `StdScalarDeserializer`.